src/keys.rs

```diff
@@ -556,7 +547,7 @@ pub struct IncomingViewingKey {
 impl From<&FullViewingKey> for IncomingViewingKey {
     fn from(fvk: &FullViewingKey) -> Self {
         IncomingViewingKey {
-            dk: fvk.into(),
+            dk: fvk.derive_dk_ovk().0,
             ivk: fvk.into(),
```

Comment:
@daira and @nuttycom would prefer not to use From for authority-changing derivations; we'll file another issue about that.